solr/core/src/java/org/apache/solr/storage/AccessDirectory.java

```diff
@@ -180,29 +180,11 @@ public void rename(String source, String dest) throws IOException {
 
   @Override
   public void sync(Collection<String> names) throws IOException {
-    ensureOpen();
-
-    for (String name : names) {
-      try {
-        fsync(name);
-      } catch (NoSuchFileException | FileNotFoundException ex) {
-        LazyEntry lazy = open(name);
-        if (lazy == null) {
-          throw ex;
-        } else {
-          try {
-            fsync(lazy.lazyName);
-          } catch (IOException ex1) {
-            // try one more time with the original name in case the file was in the
-            // process of being renamed
-            fsync(name);
-          }
-        }
-      }
-    }
-
-    // to trigger `maybeDeletePendingFiles()`
-    super.sync(Collections.emptySet());
+    super.sync(names);
```

Comment:
Since this method just calls `super.sync`, couldn't we just delete this method altogether?

Reply:
Definitely; I just left it this way to prevent any temptation to override this in the future, since it is a little counterintuitive maybe. It also provides a nice place to leave a comment inline where someone who's interested would be sure to find it.